lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -1209,3 +1217,60 @@ def get_course_run_status(course_overview, certificate_info):
             else:
                 return CourseRunProgressStatuses.UPCOMING
         return None
+
+
+class EnrollmentDataResetView(APIView):
+    """
+    Resets enrollments and users for a given organization and set of programs.
+    Note, this will remove ALL users from the input organization.
+
+    Path: ``/api/program_enrollments/v1/integration-reset/``
+
+    Accepts: [POST]
+
+    ------------------------------------------------------------------------------------
+    POST
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Enrollments and users sucessfully deleted
+        * 400: Bad Requeset - Program does not match the requested organization
+        * 401: Unauthorized - The requesting user is not authenticated.
+        * 404: Not Found - A requested program does not exist.
+
+    **Response**
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS,)
+
+    @transaction.atomic
+    def post(self, request):
+        """
+        Reset enrollment and user data for organization
+        """
+        if not settings.FEATURES.get('ENABLE_ENROLLMENT_RESET'):
+            return Response('reset not enabled on this environment', status.HTTP_501_NOT_IMPLEMENTED)
+
+        try:
+            org_key = request.data['organization']
+        except KeyError:
+            return Response("missing required body content 'organization'", status.HTTP_400_BAD_REQUEST)
+
+        try:
+            organization = Organization.objects.get(short_name=org_key)
+        except Organization.DoesNotExist:
+            return Response('organization {} not found'.format(org_key), status.HTTP_404_NOT_FOUND)
+
+        idp_slug = get_provider_slug(organization)
```

Comment:
What about an organization without any idp setup? like curtinx?

Reply:
I guess in that case we just skip this `remove_social_auth_users` command, right?

Reply:
thanks, this would have thrown an exception on curtin.  I've fixed this to just skip the command.